packages/admin/admin/src/common/buttons/feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,202 @@
+
+export function FeedbackButton(inProps: FeedbackButtonProps) {
+    const {
+        loading,
+        hasErrors,
+        children,
+        variant = "contained",
+        color = "primary",
+        classes,
+        disabled,
+        loadingIcon = <ThreeDotSaving />,
+        startIcon,
+        endIcon,
+        tooltipSuccessMessage,
+        tooltipErrorMessage,
+        slotProps,
+
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFeedbackButton",
+    });
+
+    const [displayState, setDisplayState] = React.useState<FeedbackButtonDisplayState>("idle");
+
+    const ownerState: OwnerState = {
+        displayState: displayState,
+        variant,
+        color,
+    };
+
+    const resolveIconForDisplayState = (displayState: FeedbackButtonDisplayState) => {
+        if (startIcon && displayState !== "loading") {
+            return startIcon;
+        } else if (endIcon && displayState !== "loading") {
+            return endIcon;
+        }
+        if (displayState === "loading") {
+            return loadingIcon;
+        }
+    };
+
+    const resolveTooltipForDisplayState = (displayState: FeedbackButtonDisplayState) => {
+        if (displayState === "success") {
+            return "success";
+        } else if (displayState === "fail") {
+            return "error";
+        } else {
+            return "neutral";
+        }
+    };
+
+    React.useEffect(() => {
+        let timeoutId: number | undefined;
+
+        if (displayState === "idle" && loading) {
+            setDisplayState("loading");
+        }
+
+        // Display Error
+        else if (displayState === "loading" && hasErrors) {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState("fail");
+            }, 500);
+        }
+        // Display Success
+        else if (displayState === "loading" && !loading && !hasErrors) {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState("success");
+            }, 500);
+        }
+        // Return to idle
+        else if (displayState === "fail") {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState("idle");
+            }, 5000);
+        }
+        // Return to idle
+        else if (displayState === "success") {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState("idle");
+            }, 2000);
+        }
+
+        return () => {
+            if (timeoutId !== undefined) {
+                window.clearTimeout(timeoutId);
+            }
+        };
+    }, [displayState, loading, hasErrors]);
+
+    return (
+        <Root
+            ownerState={ownerState}
+            {...slotProps}
+            {...restProps}
+            startIcon={
+                startIcon && (
+                    <SuccessTooltip
+                        title={displayState === "fail" ? tooltipErrorMessage : tooltipSuccessMessage}
```

Comment:
Why exactly can the `SuccessTooltip` show the error message, if there is a separate `ErrorTooltip`?
Same for showing the success message in the `ErrorTooltip`. 

Is it maybe not necessary to have two separate `Tooltip` components?

Reply:
Yes, that's true. The only difference between the tooltips is the placement depending on where the tooltip is used (startAdornment or endAdornment) - but otherwise the styling is the same. 

I deleted one of the Tooltips and use only one now.

Reply:
Nice 👍🏼 

If both tooltips are basically the same now, you could define a single tooltip above to prevent duplicate code:

```tsx
const tooltip = (
    <Tooltip
        title={displayState === "fail" ? tooltipErrorMessage : tooltipSuccessMessage}
        open={displayState === "fail" || displayState === "success"}
        placement={endIcon && !startIcon ? "top-end" : "top-start"}
        variant={resolveTooltipForDisplayState(displayState)}
        {...slotProps?.tooltip}
    >
        <>{resolveIconForDisplayState(displayState)}</>
    </Tooltip>
);
```